src/other/tables.jl

```diff
@@ -48,6 +48,18 @@ DataFrame!(x::Vector{<:NamedTuple}) =
                         "`$(typeof(x))` without allocating new columns: use " *
                         "`DataFrame(x)` instead"))
 
+for T in [DataFrameRows, DataFrameColumns]
+    @eval begin
```

Comment:
Do you really need a loop? Isn't `::Union{DataFrameRows, DataFrameColumns}` enough?

Reply:
Ah, I hadn't seen @bkamins's comment above. I'd just repeat the `Union` without defining a custom type alias.

Reply:
I'm OK with both approaches. I wrote this as @bkamins preferred this approach. Ref: https://github.com/JuliaData/DataFrames.jl/pull/2055#discussion_r358008369

Reply:
I am OK with both - the @eval approach is used in Base often. But using `Union` without defining the alias is also OK (I just prefer not to introduce the alias here).

Reply:
Does https://github.com/JuliaData/DataFrames.jl/pull/2055/commits/22e32dbe445bf3ce1e565cee0f56e9d2a5275197 look good?